src/reactA11yIframesRule.ts

```diff
@@ -0,0 +1,87 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getJsxAttributesFromJsxElement } from './utils/JsxAttribute';
+
+const IFRAME_ELEMENT_NAME: string = 'iframe';
+const TITLE_ATTRIBUTE_NAME: string = 'title';
+const SRC_ATTRIBUTE_NAME: string = 'src';
+const HIDDEN_ATTRIBUTE_NAME: string = 'hidden';
+const IFRAME_EMPTY_TITLE_ERROR_STRING: string = 'An iframe element must have a non-empty title.';
+const IFRAME_EMPTY_OR_HIDDEN_ERROR_STRING: string = 'An iframe element should not be hidden or empty.';
+const IFRAME_UNIQUE_TITLE_ERROR_STRING: string = 'An iframe element must have a unique title.';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'react-a11y-iframes',
+        type: 'functionality',
+        description: 'Enforce that iframe elements are not empty, have title, and are unique.',
+        options: null, // tslint:disable-line:no-null-keyword
+        optionsDescription: '',
+        typescriptOnly: false,
+        issueClass: 'Non-SDL',
+        issueType: 'Error',
+        severity: 'Important',
+        level: 'Opportunity for Excellence',
+        group: 'Accessibility'
+    };
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return sourceFile.languageVariant === ts.LanguageVariant.JSX ? this.applyWithFunction(sourceFile, walk) : [];
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>) {
+    const previousTitles: Set<string> = new Set();
+    function cb(node: ts.Node): void {
+        if (
+            tsutils.isJsxFragment(node) ||
```

Comment:
Not sure that it makes sense to reset titles list when `JsxFragment` is encountered.
`<>` can be between two `iframe`s with identical titles - which will lead to false-negative result.

If titles list should be reset on `<>` I think then special checks for `<React.Fragment>` and `<Fragment>` (depending on import in ideal case) should be added for symmetry.

What do you think?